### PR TITLE
Base.3

### PR DESCRIPTION
**Pull request by redheadedrod**
Some updates... Please check for possible collision with previous psize/tsize fix. Not sure if I undid your patch with this update?

### CODE REVIEW

File: envi/archs/arm/disasm.py [hunk 1]
```diff
@@ -1217,6 +1222,32 @@ def p_uncond(opval, va):
                 shval = (opval>>7) & 0x1f
                 olist = (ArmScaledOffsetOper(Rn, Rm, shtype, shval, va, psize=psize), )
             return (opcode, mnem, olist, 0)
+        elif (opval & 0xff000f0) == 0x5700010:
+            #clrex
+            mnem = "clrex"
+            olist =()
+            #fixme : need opcode
+            opcode = INS_BLX
```
Comment (atlas0fd00m) [inline envi/archs/arm/disasm.py]: how is this an INS_BLX?  if you're going to leave it as anything "meaningless" don't choose a branch :)
better yet, please create INS_CLREX in const.py

File: envi/archs/arm/disasm.py [hunk 2]
```diff
@@ -1217,6 +1222,32 @@ def p_uncond(opval, va):
...
+            return (opcode, mnem, olist, 0)
+        elif (opval & 0xff000e0) == 0x5700040:
+            #dmb/dsb
+            print "FIXME: need opcode"
```
Comment (atlas0fd00m) [inline envi/archs/arm/disasm.py]: same here

File: envi/archs/arm/disasm.py [hunk 3]
```diff
@@ -1217,6 +1222,32 @@ def p_uncond(opval, va):
...
+            if (opval & 0x10 )== 0x10:
+                mnem = 'dmb'
+            else:
+                mnem = 'dsb'
+            olist = (ArmBarrierOption(option),)
+            opcode = IENC_UNCOND_PLD
+            return (opcode, mnem, olist, 0)
+        elif (opval & 0xff000f0) == 0x5700060:
+            #isb
+            print "FIXME: need opcode"
```
Comment (atlas0fd00m) [inline envi/archs/arm/disasm.py]: and here :)

File: envi/archs/arm/disasm.py (hunk 1 above)
Comment (redheadedrod) [inline envi/archs/arm/disasm.py]: Will email about this... Was an issue I had that I am now reminded about

**Comment by atlas0fd00m** [comment]
please make sure the conflicts are resolved.

**Comment by atlas0fd00m** [comment]
still conflicting

File: envi/archs/arm/const.py [hunk 4]
```diff
@@ -349,6 +347,12 @@ def instrenc(encoding, index):
 INS_ASR = 85
 INS_ROR = 85
 INS_RRX = 85
+#New commands for ARMV7
```
Comment (atlas0fd00m) [inline envi/archs/arm/const.py]: lol.  we need these numbers to be unique...  i realize, that's not what's there currently.  i'll fix it later.

**Pull request merged**